### DISCUSSION TITLE
Getting found include directories for cairo

### DISCUSSION THREAD

**Discussion by andy5995** [mesonbuild/meson#11171] [Q&A] [answered]
Is there a way to get found include directories from the dependency object?

Looking through [this meson doc](https://mesonbuild.com/Reference-manual_returned_dep.html), I didn't find anything suitable.

I basically want to change the first argument of `has_header`. For example,  

```
cc.has_header('cairo/cairo-pdf.h', required : true)
```

to

```
cc.has_header(cairo_dep.include_dir + 'cairo-pdf.h', required : true)
```

https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/5325/diffs#0cc1139e3347f573ae1feee5b73dbc8a8a21fcfa_388_388

Though I'm not really familiar with cairo. Is hard-coding the include dir usually robust? Or is there just simply a better way to check for the includes?

**Comment by andy5995** [comment, marked as answer]
Ah, someone suggested I just pass the dep, like so: 

    cc.has_header('cairo-pdf.h', dependencies : cairo_dep, required : true)`

**Reply by xclaesse** [reply]
Cairo even has a different pkg-config name per backend: `dependency('cairo-pdf')` should work.

**Reply by andy5995** [reply]
> Cairo even has a different pkg-config name per backend: `dependency('cairo-pdf')` should work.

Ok, right now the code on the [merge request](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/5325) looks like:

```
if os_unix
  dependency('cairo-ps', required: true)
endif
if cairo_dep.type_name() == 'pkgconfig'
  foreach feature : ['pdf', 'svg']
     dependency('cairo-' + feature, required: true)
  endforeach
endif
```

**Reply by xclaesse** [reply]
Nitpicking:
-  `required:true` is not needed as that's the default. 
- You don't need to restrict to pkgconfig, even when Cairo is a subproject it should override those dependencies.
- you probably want something like `cairo_deps += dependency()` to build a list of deps.
